src/meta/src/model_v2/function.rs

```diff
@@ -12,34 +12,40 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use risingwave_pb::catalog::function::Kind;
+use risingwave_pb::catalog::PbFunction;
 use sea_orm::entity::prelude::*;
+use sea_orm::ActiveValue;
+
+use crate::model_v2::{DataType, DataTypeArray, FunctionId};
+
+#[derive(Clone, Debug, PartialEq, Eq, EnumIter, DeriveActiveEnum)]
+#[sea_orm(rs_type = "String", db_type = "String(None)")]
+pub enum FunctionKind {
+    #[sea_orm(string_value = "Scalar")]
+    Scalar,
+    #[sea_orm(string_value = "Table")]
+    Table,
+    #[sea_orm(string_value = "Aggregate")]
+    Aggregate,
+}
 
 #[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq)]
 #[sea_orm(table_name = "function")]
 pub struct Model {
     #[sea_orm(primary_key, auto_increment = false)]
-    pub function_id: i32,
+    pub function_id: FunctionId,
     pub name: String,
-    pub schema_id: i32,
-    pub database_id: i32,
-    pub arg_types: Option<Json>,
-    pub return_type: Option<String>,
-    pub language: Option<String>,
-    pub link: Option<String>,
-    pub identifier: Option<String>,
-    pub kind: Option<Json>,
+    pub arg_types: DataTypeArray,
+    pub return_type: DataType,
+    pub language: String,
+    pub link: String,
+    pub identifier: String,
+    pub kind: FunctionKind,
 }
 
 #[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
 pub enum Relation {
```

Comment:
May I ask what this is for? 👀

Reply:
Ah, after my quick primer to sea-arm, this seems to be generated codes.

Reply:
Yes, the relation is generated by the pk definition. It can be used to simply the code of join two tables.